src/core/sim.py

```diff
@@ -32,7 +34,15 @@ def step(self) -> PropagatedOutput:
         current_event = self.event_queue.get()
         self.state_time, self.observed_state = current_event.evaluate(self.state_time)
         # TODO: Feed outputs of sensor models into FSW and return actuator's state as part of `PropagatedOutput`
-
+        # Current implementation uses Shared Memory to pass data
+        observedArray = self.observed_state.to_array()
+        try:
+            shm = shared_memory.SharedMemory(name="Simulator Data") 
+        except (FileNotFoundError) as e:
+            log.warn("Some external processes closed the shared memory. I will restart it, but this may lead to an error on any other processes listening on this memory. Unsafe behaviour ahead!")
+            shm = shared_memory.SharedMemory(create=True, name="Simulator Data", size=getsizeof(observedArray))
+        stateArray = np.ndarray(observedArray.shape, dtype=observedArray.dtype, buffer=shm.buf)
+        stateArray[:] = observedArray[:]
```

Comment:
Any reason why you're not using `numpy.copy()` here?

Reply:
Yes, as the actual array underlying state array has to remain constant (since it refers to a specific memory buffer). ndarray also has no copy constructor, so the elements have to be copied into the array one by one